nodejs/awsx/ec2/securityGroup.ts

```diff
@@ -80,32 +96,33 @@ export class SecurityGroup extends pulumi.ComponentResource {
         name: string, id: pulumi.Input<string>,
         args: SecurityGroupArgs = {}, opts: pulumi.ComponentResourceOptions = {}) {
 
-        return new SecurityGroup(name, {
+        return SecurityGroup.create(name, {
             ...args,
             securityGroup: aws.ec2.SecurityGroup.get(name, id, {}, opts),
         }, opts);
     }
 
-    public createEgressRule(name: string, args: x.ec2.SimpleSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): x.ec2.EgressSecurityGroupRule;
-    public createEgressRule(name: string, args: x.ec2.EgressSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): x.ec2.EgressSecurityGroupRule;
+    public createEgressRule(name: string, args: x.ec2.SimpleSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): Promise<x.ec2.EgressSecurityGroupRule>;
+    public createEgressRule(name: string, args: x.ec2.EgressSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): Promise<x.ec2.EgressSecurityGroupRule>;
     public createEgressRule(name: string, args: x.ec2.SimpleSecurityGroupRuleArgs | x.ec2.EgressSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions) {
-        return new x.ec2.EgressSecurityGroupRule(name, this, args, opts);
+        return x.ec2.EgressSecurityGroupRule.create(name, this, args, opts);
     }
 
-    public createIngressRule(name: string, args: x.ec2.SimpleSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): x.ec2.IngressSecurityGroupRule;
-    public createIngressRule(name: string, args: x.ec2.IngressSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): x.ec2.IngressSecurityGroupRule;
+    public createIngressRule(name: string, args: x.ec2.SimpleSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): Promise<x.ec2.IngressSecurityGroupRule>;
+    public createIngressRule(name: string, args: x.ec2.IngressSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions): Promise<x.ec2.IngressSecurityGroupRule>;
     public createIngressRule(name: string, args: x.ec2.SimpleSecurityGroupRuleArgs | x.ec2.IngressSecurityGroupRuleArgs, opts?: pulumi.ComponentResourceOptions) {
-        return new x.ec2.IngressSecurityGroupRule(name, this, args, opts);
+        return x.ec2.IngressSecurityGroupRule.create(name, this, args, opts);
     }
 }
 
-(<any>SecurityGroup.prototype.createEgressRule).doNotCapture = true;
-(<any>SecurityGroup.prototype.createIngressRule).doNotCapture = true;
+utils.Capture(SecurityGroup.prototype).initialize.doNotCapture = true;
+utils.Capture(SecurityGroup.prototype).createEgressRule.doNotCapture = true;
+utils.Capture(SecurityGroup.prototype).createIngressRule.doNotCapture = true;
```

Comment:
new pattern for marking functions as non-capturable that doesn't require casting to `<any>`